### ISSUE TITLE
Maven warning when using Flyway plugin with parallel builds

### ISSUE THREAD

**Issue by cowwoc** [flyway/flyway#1305]
###### What version of Flyway are you using?

4.0
###### What database are you using (type & version)?

Doesn't matter
###### What operating system are you using?

Windows 7
###### What did you do?

(Please include the content causing the issue, any relevant configuration settings, and the command you ran)

```
            <plugin>
                <groupId>org.flywaydb</groupId>
                <artifactId>flyway-maven-plugin</artifactId>
                <version>4.0</version>
                <executions>
                    <execution>
                        <id>migrate-db</id>
                        <goals>
                            <goal>migrate</goal>
                        </goals>
                        <phase>generate-sources</phase>
                    </execution>
                </executions>
                <dependencies>
                    <dependency>
                        <groupId>mysql</groupId>
                        <artifactId>mysql-connector-java</artifactId>
                        <version>${mysql.version}</version>
                    </dependency>
                </dependencies>
            </plugin>
```

Invoked "mvn clean install -T 8" to trigger parallel builds.
###### What did you expect to see?

According to https://flywaydb.org/documentation/faq#parallel Flyway supports parallel migrations, so I am not expecting to see any warnings.
###### What did you see instead?

```
[WARNING] *****************************************************************
[WARNING] * Your build is requesting parallel execution, but project      *
[WARNING] * contains the following plugin(s) that are not marked as       *
[WARNING] * @threadSafe to support parallel building.                     *
[WARNING] * While this /may/ work fine, please look for plugin updates    *
[WARNING] * and/or request plugins be made thread-safe.                   *
[WARNING] * If reporting an issue, report it against the plugin in        *
[WARNING] * question, not against maven-core                              *
[WARNING] *****************************************************************
[WARNING] The following plugins are not marked @threadSafe in data-access:
[WARNING] org.flywaydb:flyway-maven-plugin:4.0
[WARNING] *****************************************************************
```

**Comment by axelfontaine** [comment]
The word parallel in this FAQ entry referred to clusters of machines and not Maven builds.

However I agree that having the ability to use Maven's parallel builds is a valuable feature.

**Comment by axelfontaine** [comment]
This has been fixed as part of the Maven 3.0 conversion of the plugin.
